src/sentry/web/frontend/pipeline_advancer.py

```diff
@@ -40,15 +38,26 @@ def handle(self, request: Request, provider_id: str) -> HttpResponseBase:
             if pipeline:
                 break
 
+        # GitHub apps may be installed directly from GitHub, in which case
+        # they will redirect here *without* being in the pipeline. If that happens
+        # redirect to the integration install org picker.
         if (
-            provider_id in FORWARD_INSTALL_FOR
+            provider_id == "github"
             and request.GET.get("setup_action") == "install"
             and pipeline is None
         ):
             installation_id = request.GET.get("installation_id")
-            return self.redirect(
+            rv = self.redirect(
                 reverse("integration-installation", args=[provider_id, installation_id])
             )
+            rv.set_cookie(
```

Comment:
Is there an advantage/requirement here to use `set_cookie` versus the `request.session`? The session is signed so we will be able to prevent tampering.